src/dmd/mars.d

```diff
@@ -794,6 +694,121 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
     return status;
 }
 
+/**
+ * Parses the command line arguments and configuration files
+ *
+ * Params:
+ *   argc = Number of arguments passed via command line
+ *   argv = Array of string arguments passed via command line
+ *   params = parametes from argv
+ *   files = files from argv
+ *  Returns: true on faiure
+ */
+version(NoMain) {} else
+bool parseCommandlineAndConfig(size_t argc, const(char)** argv, ref Param params, ref Strings files)
+{
+    // Detect malformed input
+    static bool badArgs()
+    {
+        error(Loc.initial, "missing or null command line arguments");
+        return true;
+    }
+
+    if (argc < 1 || !argv)
+        return badArgs();
+    // Convert argc/argv into arguments[] for easier handling
+    Strings arguments = Strings(argc);
+    for (size_t i = 0; i < argc; i++)
+    {
+        if (!argv[i])
+            return badArgs();
+        arguments[i] = argv[i];
+    }
+    if (!responseExpand(arguments)) // expand response files
+        error(Loc.initial, "can't open response file");
+    //for (size_t i = 0; i < arguments.dim; ++i) printf("arguments[%d] = '%s'\n", i, arguments[i]);
+    files.reserve(arguments.dim - 1);
+    // Set default values
+    params.argv0 = arguments[0].toDString;
+
+    // Temporary: Use 32 bits OMF as the default on Windows, for config parsing
+    static if (TARGET.Windows)
+    {
+        params.is64bit = false;
+        params.mscoff = false;
+    }
+
+    version (Windows)
+        enum iniName = "sc.ini";
+    else version (Posix)
+        enum iniName = "dmd.conf";
+    else
+        static assert(0, "fix this");
+
+    global.inifilename = parse_conf_arg(&arguments);
+    if (global.inifilename)
+    {
+        // can be empty as in -conf=
+        if (global.inifilename.length && !FileName.exists(global.inifilename))
+            error(Loc.initial, "Config file '%.*s' does not exist.",
+                  cast(int)global.inifilename.length, global.inifilename.ptr);
+    }
+    else
+        global.inifilename = findConfFile(params.argv0, iniName);
```

Comment:
DMD style is to use brackets for the `else` branch also, if the `if` branch does, even if it's only one line.